pkgs/by-name/ni/nixfmt-tree/package.nix

```diff
@@ -122,49 +121,63 @@ treefmtWithConfig.overrideAttrs {
     platforms = lib.platforms.all;
   };
 
-  passthru.tests.simple = runCommand "nixfmt-tree-test-simple" { } ''
-    export XDG_CACHE_HOME=$(mktemp -d)
-    cat > unformatted.nix <<EOF
-    let to = "be formatted"; in to
-    EOF
-
-    cat > formatted.nix <<EOF
-    let
-      to = "be formatted";
-    in
-    to
-    EOF
-
-    mkdir -p repo
-    (
-      cd repo
-      mkdir .git dir
-      touch .git/index
-      cp ../unformatted.nix a.nix
-      cp ../unformatted.nix dir/b.nix
-
-      ${lib.getExe nixfmt-tree} dir
-      if [[ "$(<dir/b.nix)" != "$(<../formatted.nix)" ]]; then
-        echo "File dir/b.nix was not formatted properly after dir was requested to be formatted"
-        exit 1
-      elif [[ "$(<a.nix)" != "$(<../unformatted.nix)" ]]; then
-        echo "File a.nix was formatted when only dir was requested to be formatted"
-        exit 1
-      fi
-
-      (
-        cd dir
-        ${lib.getExe nixfmt-tree}
-      )
-
-      if [[ "$(<a.nix)" != "$(<../formatted.nix)" ]]; then
-        echo "File a.nix was not formatted properly after running treefmt without arguments in dir"
-        exit 1
-      fi
-    )
-
-    echo "Success!"
-
-    touch $out
-  '';
+  passthru.tests.simple =
+    runCommand "nixfmt-tree-test-simple"
+      {
+        nativeBuildInputs = [
+          git
+          nixfmt-tree
+          writableTmpDirAsHomeHook
+        ];
+      }
+      ''
+        git config --global user.email "nix-builder@nixos.org"
+        git config --global user.name "Nix Builder"
+
+        cat > unformatted.nix <<EOF
+        let to = "be formatted"; in to
+        EOF
+
+        cat > formatted.nix <<EOF
+        let
+          to = "be formatted";
+        in
+        to
+        EOF
+
+        mkdir -p repo
+        (
+          cd repo
+          mkdir dir
+          cp ../unformatted.nix a.nix
+          cp ../unformatted.nix dir/b.nix
+
+          git init
+          git add .
+          git commit -m "Initial commit"
+
+          treefmt dir
```

Comment:
This is now failing with:

> Error: path dir not inside the tree root /nix/store

However, I don't understand why treefmt things the tree-root is `/nix/store`:

`pwd`, `realpath .`, and `git rev-parse --show-toplevel` all print `/build/repo`.

Am I missing something @brianmcgee ?

Reply:
I tried reverting https://github.com/numtide/treefmt/commit/a22a307ea057d628b51f71b6be2daec802365542 just to be sure, and still got the same behaviour. So I guess that isn't related.

Reply:
If I explicitly set `walk = "git"`, it works as intended.

So I guess there's an issue with setting the default `walk`?

Isn't it supposed to default to `"git"` when in a git repo? I can't see a default [in the docs](https://treefmt.com/latest/getting-started/configure/#walk) though.

Reply:
`auto` is the default `--walk` value, where it tries `git` and falls back to `filesystem`. 

https://github.com/numtide/treefmt/blob/60f791c3eab687018be002e470c8c119cc55c275/walk/walk.go#L205-L212

The code for detecting a git repo is here:

https://github.com/numtide/treefmt/blob/60f791c3eab687018be002e470c8c119cc55c275/walk/git.go#L146-L153

Reply:
When `--walk=auto` (default) how does this line behave? https://github.com/numtide/treefmt/blob/60f791c3eab687018be002e470c8c119cc55c275/config/config.go#L294

Does if evaluate to `false` because `"auto" != "git"`? Or does `"auto"` get coerced to whatever value it ends up _actually_ being?

Reply:
Yeah, it's not being coerced yet, which is why you needed to set `walk` to `git`. 

I need to have a think about the correct way to handle things when `walk` is `auto`.